lib/redis/namespace.rb

```diff
@@ -469,9 +483,10 @@ def add_namespace(key)
 
       case key
       when Array
-        key.map {|k| add_namespace k}
+        key.map! {|k| add_namespace k}
       when Hash
-        Hash[*key.map {|k, v| [ add_namespace(k), v ]}.flatten]
+        key.keys.each {|k| key[add_namespace(k)] = key.delete(k)}
```

Comment:
That line is modifying the `key` hash in-place. To do so it deletes the existing entry for `k` and reinserts its value `v` under the new key `add_namespace(k)`. The behaviour of `add_namespace(key)` has been modified to work in-place on its arguments, and not return new hashes or arrays. This was done to centralize the argument copying into a single pass by calling `clone_args`.